CRM/Core/BAO/CustomValue.php

```diff
@@ -230,36 +232,28 @@ public function addSelectWhereClause(string $entityName = NULL, int $userId = NU
   }
 
   /**
-   * Special checkAccess function for multi-record custom pseudo-entities
-   *
-   * @param string $entityName
-   *   APIv4-style entity name e.g. 'Custom_Foobar'
-   * @param string $action
-   * @param array $record
-   * @param int $userID
-   *   Contact ID of the active user (whose access we must check). 0 for anonymous.
-   * @return bool
-   *   TRUE if granted. FALSE if prohibited. NULL if indeterminate.
+   * Access check for multi-record custom pseudo-entities
+   * @see \Civi\Api4\Utils\CoreUtil::checkAccessRecord
    */
-  public static function _checkAccess(string $entityName, string $action, array $record, int $userID): ?bool {
+  public static function on_civi_api4_authorizeRecord(AuthorizeRecordEvent $e): void {
+    $groupName = \Civi\Api4\Utils\CoreUtil::getCustomGroupName($e->getEntityName());
+    if (!$groupName) {
+      return;
+    }
+
```

Comment:
Yes that's what it's doing. I think you understand it. This is doing ACL checks for the pseudo-entities that exist for every multi-record custom group. Their API names all start with `Custom_` (and all concrete classes do not, so that's how we tell them apart).